### ISSUE TITLE
Call for proposals on lambda notation

### ISSUE THREAD

**Issue by jonsterling** [RedPRL/redtt#284]
The current notation for lambda abstractions (used for both functions and paths) is `lam x -> e`, supporting unicode-equivalents of `lam` and `->`. I'm not a huge fan of this notation for a couple reasons.

1. For things like case branches we use `=>`. Maybe we should use that here too.
2. `lam` is too wordy; maybe we should also support `\` for ascii users.
3. Maybe some completely different notation is more immortal.

What do folks think? /cc @favonia  @cangiuli @mortberg @ecavallo 

I'd like to choose whatever makes (1) code easiest to read, (2) code least frustrating to write.

**Comment by favonia** [comment]
Is there even a reason to have both `=>` and `->`? Will people be confused if we use `->` everywhere?

**Comment by jonsterling** [comment]
I lean toward your point of view too, though personally I am OK with using -> for things like function types.

**Comment by favonia** [comment]
Will you be satisfied with a simple PR that 1. replace all `=>` with `->` and 2. add `\`?

**Comment by jonsterling** [comment]
@favonia I think that's reasonable
